pytorch_lightning/strategies/fully_sharded_native.py

```diff
@@ -0,0 +1,268 @@
+            backward_prefetch: (Optional[BackwardPrefetch]):
+                This is an experimental feature that is subject to change in the
+                the near future. It allows users to enable two different backward_prefetch
+                algorithms to help backward communication and computation overlapping.
+                Pros and cons of each algorithm is explained in the class ``BackwardPrefetch``.
+        """
+        if not _TORCH_GREATER_EQUAL_1_11:
+            raise MisconfigurationException(
+                "`DDPFullyShardedNativeStrategy` is supported from pytorch v1.11.0 onwards."
+            )
+
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+        self._process_group = None
+        self.num_processes = len(self.parallel_devices) if self.parallel_devices is not None else 0
+        self._process_group_backend: Optional[str] = process_group_backend
+        self.cpu_offload: Optional[CPUOffload] = cpu_offload
+        self.backward_prefetch: Optional[BackwardPrefetch] = backward_prefetch
+
+    @property
+    def root_device(self) -> torch.device:
+        return self.parallel_devices[self.local_rank]
+
+    @property
+    def process_group(self) -> Optional[ProcessGroup]:
+        if self._process_group is None:
+            # The plugin should have already initilized ddp in setup_environment()
+            self._process_group = _get_default_group()
+        return self._process_group
+
+    @property
+    def setup_optimizers_in_pre_dispatch(self) -> bool:
+        # Setup optimizers after the Fully Sharded Model has been made
+        return True
+
+    @property
+    def process_group_backend(self) -> Optional[str]:
+        return self._process_group_backend
+
+    def setup_environment(self) -> None:
+        self.setup_distributed()
+        super().setup_environment()
+
+    def setup_distributed(self) -> None:
+        if not self.root_device.type == "cuda":
+            raise MisconfigurationException(
+                f"You selected strategy to be `{self.strategy_name}`, but GPU is not available."
+            )
+        reset_seed()
+        # set warning rank
+        rank_zero_only.rank = self.global_rank
+
+        self._process_group_backend = self._get_process_group_backend()
+        init_dist_connection(self.cluster_environment, self._process_group_backend)
+
+    def _get_process_group_backend(self) -> str:
+        return (
+            self._process_group_backend
+            or _get_process_group_backend_from_env()
+            or get_default_process_group_backend_for_device(self.root_device)
+        )
+
+    def setup(self, trainer: "pl.Trainer") -> None:
+        self.accelerator.setup(trainer)
+
+        if trainer.state.fn == TrainerFn.FITTING and self._layer_sync:
```

Comment:
is sync batch norm supported by torch FSDP?

Reply:
It should be supported, although there is some follow up work needed to be done for mixed precision, although this version of the plugin I guess won't have access to mixed precision, as that will ship in PyTorch 1.12. Are there any issues we're seeing at the moment?

Reply:
Ok so then SyncNormBatch should just work. Correct, mixed precision is marked as not supported for this strategy. No issues, but trying to get to similar level of parity with FSDP V1 (from fairscale) plugin.

Reply:
I've added in a test with sync batch norm as well. And its passing the test.